python_modules/dagster/dagster/_utils/calculate_data_time.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from typing import AbstractSet, Mapping, Optional, cast
+
+from dagster import AssetKey, DagsterEventType, DagsterInstance, EventLogRecord, EventRecordsFilter
+
+UPSTREAM_MATERIALIZATION_TIMES_TAG = ".dagster/upstream_times"
+
+
+def get_upstream_materialization_times_for_record(
+    instance: DagsterInstance,
+    upstream_asset_key_mapping: Mapping[str, AbstractSet[str]],
```

Comment:
With the changes in [this PR](https://github.com/dagster-io/dagster/pull/10402), can we operate on asset keys instead of strings?